src/auto-compounder/AutoCompounderViews.sol

```diff
@@ -0,0 +1,114 @@
+import { QuoteExactOutputSingleParams } from "./interfaces/IQuoter.sol";
+import { UniswapV3Logic } from "./libraries/UniswapV3Logic.sol";
+
+/**
+ * @title Off-chain view functions for UniswapV3 AutoCompounder Asset-Manager.
+ * @author Pragma Labs
+ * @notice This contract holds view functions accessible for initiators to check if the fees of a certain Liquidity Position can be compounded.
+ */
+contract AutoCompounderViews {
+    using FixedPointMathLib for uint256;
+
+    /* //////////////////////////////////////////////////////////////
+                            CONSTANTS
+    ////////////////////////////////////////////////////////////// */
+
+    // The contract address of the Asset Manager.
+    IAutoCompounder public immutable autoCompounder;
+
+    /* //////////////////////////////////////////////////////////////
+                            CONSTRUCTOR
+    ////////////////////////////////////////////////////////////// */
+
+    /**
+     * @param autoCompounder_ The contract address of the Asset-Manager for compounding UniswapV3 fees of a certain Liquidity Position.
+     */
+    constructor(address autoCompounder_) {
+        autoCompounder = IAutoCompounder(autoCompounder_);
+    }
+
+    /* ///////////////////////////////////////////////////////////////
+                      OFF-CHAIN VIEW FUNCTIONS
+    /////////////////////////////////////////////////////////////// */
+
+    /**
+     * @notice Off-chain view function to check if the fees of a certain Liquidity Position can be compounded.
+     * @param id The id of the Liquidity Position.
+     * @return isCompoundable_ Bool indicating if the fees can be compounded.
+     * @dev While this function does not persist state changes, it cannot be declared as view function.
+     * Since quoteExactInputSingle() of Uniswap's Quoter02.sol uses a try - except pattern where it first
+     * does the swap (with state changes), next it reverts (state changes are not persisted) and information about
+     * the final state is passed via the error message in the expect.
+     */
+    function isCompoundable(uint256 id) external returns (bool isCompoundable_) {
+        // Fetch and cache all position related data.
+        IAutoCompounder.PositionState memory position = autoCompounder.getPositionState(id);
+
+        // Check that pool is initially balanced.
+        // Prevents sandwiching attacks when swapping and/or adding liquidity.
+        if (autoCompounder.isPoolUnbalanced(position)) return false;
+
+        // Get fee amounts
+        IAutoCompounder.Fees memory fees;
+        (fees.amount0, fees.amount1) = UniswapV3Logic._getFeeAmounts(id);
+
+        // Total value of the fees must be greater than the threshold.
+        if (autoCompounder.isBelowThreshold(position, fees)) return false;
+
+        // Remove initiator reward from fees, these will be send to the initiator.
+        uint256 initiatorShare = autoCompounder.INITIATOR_SHARE();
```

Comment:
Maybe we should pass autoCompounder as input (if we have multiple auto compounder with different settings?).
If we keep it hardcoded, autoCompounder should be in capital letters